KomaMRIFiles/src/Sequence/Pulseq.jl

```diff
@@ -706,10 +775,47 @@ function get_block(obj, i)
     #DUR
     D = Float64[max(obj["blockDurations"][i], dur(Gx), dur(Gy), dur(Gz), dur(R[1]), dur(A[1]))]
 
-    #Extensions
-    E = Dict{String, Any}()#read_Extension(obj["extensionLibrary"], iext, i)
+     #Extensions
+     E = read_extension(obj["extensionLibrary"], obj["extensionType"],obj["triggerLibrary"],obj["labelsetLibrary"],
+     obj["labelincLibrary"],iext)
+     #E = [Vector{Extension}[]]#read_Extension(obj["extensionLibrary"], iext, i)
+
+    # Definitition
+    DEF = Dict{String,Any}()
 
     #Sequence block definition
-    s = Sequence(G,R,A,D,E)
+    s = Sequence(G,R,A,D,E,DEF)
     s
 end
+
+function read_extension(extensionLibrary,extensionType,triggerLibrary,labelsetLibrary,labelincLibrary,i)
+    EXT = [Extension[]]
+
+    if isempty(extensionLibrary) || i==0
+        return EXT
+    end
+
+    # type ref next_id
+    # next_id of 0 terminates the list
+    type, ref, next_id = extensionLibrary[i]["data"]
+
+    while true
+        if extensionType[type]["data"] == "LABELSET"
+            push!(EXT[1],LabelSet(labelsetLibrary[ref]["data"]...))
+        elseif extensionType[type]["data"] == "LABELINC"
+            push!(EXT[1],LabelInc(labelincLibrary[ref]["data"]...))
+        elseif extensionType[type]["data"] == "TRIGGERS"
+            push!(EXT[1],Trigger(triggerLibrary[ref]["data"]...))
+        else
+            @warn "Extension type not implemented"
+        end
+
+        if next_id == 0
+            break
+        else
+            type, ref, next_id = extensionLibrary[next_id]["data"]
+        end
+    end
```

Comment:
It would be nice to do this with multiple dispatch, as extensions are meant to be extended. So if the user creates a new extension called "MyEXT" they can just define maybe `read_extenstion(..., Val("MyEXT"))` or other functions like that, so they don't need to modify the code.